src/dialogue/GameNarrative.java

```diff
@@ -7,14 +7,21 @@
 
 public class GameNarrative {
     private Map<Room, Narrative> gameNarrative = new HashMap<>();
-    private boolean isNarrated = false;
+
+    public GameNarrative() {
+
+    }
+
+    private GameNarrative(Room room) {
+        gameNarrative.put(room, new Narrative());
+    }
 
     public Narrative getNarrative(Room room) {
-        return gameNarrative.get(room);
+        if (gameNarrative.containsKey(room)) return gameNarrative.get(room);
+        return new GameNarrative(room).getNarrative(room);
```

Comment:
Why create a new `GameNarative`?

Reply:
Since some rooms don't have a Narrative, it will return null. So my solution for the null return is creating a new instance of a GameNarrative with the room not listed in the current GameNarrative. The new GameNarrative then passes an empty Narrative for that room search. I think using getOrDefault would also be good here.